main.cpp

```diff
@@ -208,22 +208,23 @@ class LedResource {
     Thread blinky_thread;
     BlinkArgs *blink_args;
     void do_blink() {
-        // blink the LED
-        red_led = !red_led;
-        // up the position, if we reached the end of the vector
-        if (blink_args->position >= blink_args->blink_pattern.size()) {
-            // send delayed response after blink is done
-            M2MObjectInstance* inst = led_object->object_instance();
-            M2MResource* led_res = inst->resource("5850");
-            led_res->send_delayed_post_response();
-            red_led = 1;
-            status_ticker.attach_us(blinky, 250000);
-            return;
+        for(;;){
```

Comment:
Please use the same coding style as in rest of file, i.e. add space between "for" and "(" and before "{".